raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -307,31 +328,12 @@ private void fireHandleResign(int epoch) {
     }
 
     @Override
-    public void initialize(RaftConfig raftConfig) throws IOException {
```

Comment:
Hmm.. This is a little different than what I had in mind. I thought we would leave `initialize(RaftConfig)` as it was and move the block below to update the endpoints into `RaftManager`. From a high level, we are saying that it is the responsibility of `RaftManager` to initialize the endpoints and not `RaftClient`, which is a little more convenient for testing. Also, I had hoped to keep the call to `QuorumState.initialize` here.

Reply:
So, these changes aren't final yet and I was attempting to keep the RaftConfig and the RaftClient consistent with the endpoint updates (I.e. Avoid calls to update endpoints with the RaftConfig and with the RaftClient). From a testing perspective I see your point.

For the `QuorumState.initialize()`, I agree.

Reply:
I do think it's nice that we can make `RaftConfig` final now by the way. What if we did the following?

1. Keep `RaftConfig` in constructor
2. Move call to `QuorumState.initialize` to `KafkaRaftClient.initialize`
3. Move `updateEndpoint` calls up to `RaftManager.

Reply:
Yup. I like that too. RaftConfig being final again was definitely a goal considering we only need to have the endpoints updated in any given case.